src/plugins/sockets/connectivityEvents.js

```diff
@@ -20,7 +20,7 @@ export async function handleConnect() {
       const gameStateIndex = gameHistoryStore.currentGameStateIndex;
       const response = await gameStore.requestGameState(gameId, gameStateIndex);
       if (response?.victory?.gameOver && response.game.rematchGame) {
-        await gameStore.requestGameState(response.game.rematchGame);
+        await gameStore.requestGameState(response.game.rematchGame, -1, null, true);
```

Comment:
I don't love the complicated call signature of this function or the addition of this new arg for this specific case, but ultimately it's not that bad and this will fix the UX issue